PowerFGT/Public/cmdb/firewall/vipgroup.ps1

```diff
@@ -0,0 +1,592 @@
+#
+# Copyright 2019, Alexis La Goutte <alexis dot lagoutte at gmail dot com>
+# Copyright 2019, Benjamin Perrier <ben dot perrier at outlook dot com>
+#
+# Contribution by Brett Pound <brett underscore pound at hotmail dot com>, March 2020
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+function Add-FGTFirewallVipGroup {
+
+    <#
+        .SYNOPSIS
+        Add a FortiGate VIP Group
+
+        .DESCRIPTION
+        Add a FortiGate VIP Group
+
+        .EXAMPLE
+        Add-FGTFirewallVipGroup -name MyVipGroup -member MyVip1 -interface wan1
+
+        Add VIP Group with member MyVip1, associated to interface wan1
+
+        .EXAMPLE
+        Add-FGTFirewallVipGroup -name MyVipGroup -member MyVip1, MyVip2 -interface wan1
+
+        Add VIP Group with members MyVip1 and MyVip2, associated to interface wan1
+
+        .EXAMPLE
+        Add-FGTFirewallVipGroup -name MyVipGroup -member MyVip1 -comment "My VIP Group" -interface wan1
+
+        Add VIP Group with member MyVip1 and a comment, associated to interface wan1
+    #>
+
+    Param(
+        [Parameter (Mandatory = $true)]
+        [string]$name,
+        [Parameter (Mandatory = $true)]
+        [string[]]$member,
+        [Parameter (Mandatory = $true)]
+        [string[]]$interface,
+        [Parameter (Mandatory = $false)]
+        [ValidateLength(0, 255)]
+        [string]$comment,
+        [Parameter (Mandatory = $false)]
+        [boolean]$visibility,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('vdom') ) {
+            $invokeParams.add( 'vdom', $vdom )
+        }
+
+        if ( Get-FGTFirewallVipGroup @invokeParams -name $name -connection $connection) {
+            Throw "Already a VipGroup object using the same name"
+        }
+
+        $uri = "api/v2/cmdb/firewall/vipgrp"
+
+        $vipgrp = new-Object -TypeName PSObject
+
+        $vipgrp | add-member -name "name" -membertype NoteProperty -Value $name
+
+        #Add member to Member Array
+        $members = @( )
+        foreach ( $m in $member ) {
+            $member_name = @{ }
+            $member_name.add( 'name', $m)
+            $members += $member_name
+        }
+        $vipgrp | add-member -name "member" -membertype NoteProperty -Value $members
+
+        if ( $PsBoundParameters.ContainsKey('interface') ) {
```

Comment:
if interface is really needed, you need to remove this if

Reply:
OK, but functionally it works, because there always will be an interface, this tests true always.

Reply:
(so a more correct statement would be "if interface is really needed, you should remove this redundant if")

Reply:
amendment complete